platforms/core-configuration/encryption-services/src/main/kotlin/org/gradle/internal/encryption/impl/KeyStoreKeySource.kt

```diff
@@ -122,11 +122,16 @@ class KeyStoreKeySource(
     fun PersistentCache.keyStoreFile(): File =
         File(baseDir, "gradle.keystore")
 
+    private
     companion object {
-        // JKS does not support non-PrivateKeys
-        const val KEYSTORE_TYPE = "pkcs12"
-        private val KEYSTORE_PASSWORD = charArrayOf('c', 'c')
+        // the known type we should fall back to if the JVM's default is unsupported
+        const val FALLBACK_TYPE = "pkcs12"
```

Comment:
💭  It would be good to explicitly capture why we choose this specific fallback type